compiler/rustc_interface/src/util.rs

```diff
@@ -122,11 +124,19 @@ pub(crate) fn run_in_thread_pool_with_globals<F: FnOnce() -> R + Send, R: Send>(
         .acquire_thread_handler(jobserver::acquire_thread)
         .release_thread_handler(jobserver::release_thread)
         .num_threads(threads)
-        .deadlock_handler(|| {
+        .deadlock_handler(move || {
             // On deadlock, creates a new thread and forwards information in thread
             // locals to it. The new thread runs the deadlock handler.
-            let query_map =
-                FromDyn::from(tls::with(|tcx| QueryCtxt::new(tcx).collect_active_jobs()));
+            let query_map = FromDyn::from(
+                Weak::upgrade(compiler.read().unwrap().as_ref().unwrap()).unwrap().enter(
```

Comment:
This doesn't work as `Compiler::enter` creates a new separate `GlobalCtxt`.

Reply:
whoops, good point. Fixed now